run_training.py

```diff
@@ -1342,106 +858,109 @@ def concat_relations(relations, offsets):
+                           margin=margin,
+                           device=device).to(device)
+        model = train(model,
+                      dataset,
+                      output_dir=model_dir,
+                      relation_mat=relation_mat,
+                      mask=masks,
+                      n_epochs=n_epochs,
+                      lr=learn_rate,
+                      batch_size=batch_size,
+                      device=device,
+                      transform=True,
+                      val_split_ratio=val_split_ratio,
+                      patience=patience,
+                      )
+    else:
+        train_set, train_labels, val_set, val_labels = \
+            train_val_split(dataset, labels, val_split_ratio=val_split_ratio, seed=0)
+        tri_train_set = TripletDataset(train_labels, lambda index: augment_img(train_set[index]), n_pos_samples)
+        tri_val_set = TripletDataset(val_labels, lambda index: augment_img(val_set[index]), n_pos_samples)
+        # Data Loader
+        train_loader = DataLoader(tri_train_set,
+                                    batch_size=batch_size_adj,
+                                    shuffle=True,
+                                    num_workers=2,
+                                    pin_memory=False,
+                                    )
+        val_loader = DataLoader(tri_val_set,
+                                  batch_size=batch_size_adj,
+                                  shuffle=False,
+                                  num_workers=2,
+                                  pin_memory=False,
+                                  )
+        tri_loss = AllTripletMiner(margin=margin).to(device)
+        # tri_loss = HardNegativeTripletMiner(margin=margin).to(device)
+        ## Initialize Model ###
+
+        model = EncodeProject(arch=network, loss=tri_loss, num_inputs=num_inputs).to(device)
+
+        if start_model_path:
+            print('Initialize the model with state {} ...'.format(start_model_path))
+            model.load_state_dict(t.load(start_model_path))
+        model = train_with_loader(model,
+                              train_loader=train_loader,
+                              val_loader=val_loader,
+                              output_dir=model_dir,
+                              n_epochs=n_epochs,
+                              lr=learn_rate,
+                              device=device,
+                              patience=patience,
+                              earlystop_metric=earlystop_metric,
+                              retrain=retrain,
+                              log_step_offset=start_epoch)
+
+def parse_args():
+
+    parser = argparse.ArgumentParser()
+
+    parser.add_argument(
+        '-c', '--config',
+        type=str,
+        required=True,
+        help='path to yaml configuration file'
+    )
+
+    return parser.parse_args()
+
+
+if __name__ == '__main__':
+    args = parse_args()
+    main(args.config)
+
```

Comment:
Overall I like `run_training.py`.  Later we should look into breaking out the cli parsing commands here from the torch-specific stuff like training and dataloading.

Reply:
I agree. `main` can use some more refactoring.